### PR TITLE
chore(client): cookie handling for load balancer

### PR DESCRIPTION
**Pull request by samuelmasse**
Adds handling in the messaging client to use the bp_messaging cookie for load balancing

Closes DEV-2400

### CODE REVIEW

**Comment by samuelmasse** [comment]
Note: we'll need to release a new version of the messaging client and update it on the runtime to see the change in behavior

File: packages/client/src/base.ts [hunk 1]
```diff
@@ -72,8 +73,32 @@ export abstract class MessagingChannelBase extends Emitter<{
...
+  }
+
+  private saveCookie(cookieHeader: string[] | undefined) {
+    if (!cookieHeader) {
+      return
+    }
+
+    for (const strCookie of cookieHeader) {
+      const resCookie = cookie.parse(strCookie)
+      if (resCookie.bp_messaging) {
```
Comment (michaelmass) [inline packages/client/src/base.ts:98]: The name of the `cookie` should be extracted in a configuration setting this way it is going to be easier to change it without having to rebuild a new image of the runtime.

```
const config = {
   [...]
   messagingSessionCookie: process.env.MESSAGING_SESSION_COOKIE ?? 'bp_messaging'
}
```

**Review by michaelmass — CHANGES_REQUESTED** [review]

File: packages/client/src/base.ts [hunk 2]
```diff
@@ -72,8 +82,32 @@ export abstract class MessagingChannelBase extends Emitter<{
...
+
+  private saveCookie(cookieHeader: string[] | undefined) {
+    if (!this.sessionCookieName || !cookieHeader) {
+      return
+    }
+
+    for (const strCookie of cookieHeader) {
+      const resCookie = cookie.parse(strCookie)
+      if (resCookie[this.sessionCookieName]) {
+        this.http.defaults.headers.common['cookie'] = `${this.sessionCookieName}=${resCookie.bp_messaging};`
```
[the remark this answers is not in the document]
Comment (samuelmasse) [reply packages/client/src/base.ts]: Yup good catch

File: packages/client/src/options.ts [hunk 3]
```diff
@@ -12,6 +12,8 @@ export interface MessagingChannelOptions {
   axios?: Omit<AxiosRequestConfig, 'baseURL'>
   /** Optional logger interface that can be used to get better debugging */
   logger?: Logger
+  /** Name of the cookie for sticky sessions */
+  sessionCookieName?: string
 }
 
 export interface MessagingOptions extends Omit<MessagingChannelOptions, 'adminKey'>, MessagingClientAuth {
```
Comment (samuelmasse) [inline packages/client/src/options.ts:19]: @michaelmass Added the sessionCookieName config

**Review by michaelmass — APPROVED** [review]
LGTM

**Pull request merged**